### PR TITLE
tommy's gsoc intro

### PR DESCRIPTION
**Pull request by tommyli3318**

### CODE REVIEW

File: posts/tommyli3318-intro.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Hi all! :)
+
+My name is Taofeng Li and I go by the nickname of Tommy, gihub handle `tommyli3318 <https://github.com/tommyli3318>`_. I'm a Computer Science major at the University of California, Irvine in the United States. I am very excited to announce that I'll be contributing to Nuitka this summer under the Google Summer of Code program!
```
Comment (kayhayen) [inline posts/tommyli3318-intro.rst]: `gihub` -> `github`

**Comment by tommyli3318** [comment]
Oh yeah, I wanted to ask this before, as Im still new to working on big projects like Nuitka. One thing that is annoying is merge conflicts, such as the one I'm getting right now. Is there a way to pull from Nuitka/Nuitka-website?

**Comment by kayhayen** [comment]
@tommyli3318 you ought to always do `git pull --rebase` when you are on `master` and `develop`. For your feature branch, I would normally avoid merges, and instead do `git rebase origin/develop`. Then, when there is an amended commit or something, what can happen is that your old version of that commit conflicts with the amended one. I changed @bksahu posting date, and I like will yours, so this happens.

Also happens in Nuitka main repo, but is very rare, but will happen to rescue `git bisect` abilities.

When it happens, normally you should get away with `git rebase --skip` if the changes are not at all yours, then after resolving the merge conflict, best case you wouldn't have a change. Skipping is of course easiest then. Otherwise, you do a git mergetool, where I recommend kdiff3 or meld, but I think the first one is best, where you can do code voting ("A", "B", "C"), and edit directly in the result as it needs to be. Then exit, and `git rebase --continue`.

Merge commits have several drawbacks, one of which is that merge conflict resolutions become special part of the commit, that doesn't even show up in standard `git log -p`, but I think you need to remember to add `-cc` or something to make it show those.

Therefore I recommend to always be fast forward in your pull requests.

Also, what I am going to do normally is to squash merge, i.e. the commit messages and individual commits do not survive anyway. Fixups of broken things, are therefore something you can do, but they later get squashed, so each feature change becomes one commit of yours (your authorship remains intact).

It would be asked too much I believe to have to produce perfect code and perfect commit history, where each commit works separately and perfectly. 

Now I need to remember to add that to the developer manual of Nuitka.

**Comment by kayhayen** [comment]
Thanks a lot, I will push it out for rending during the day.

**Pull request merged**